### ISSUE TITLE
`sourceName` input is altered, apparently based on `outputName` input

### ISSUE THREAD

**Issue by jmm** [google/traceur-compiler#1695]
With traceur 0.0.83, this:

```js
  (new traceur.NodeCompiler({sourceMaps: 'inline'}))
    .compile(';', '/a/b/c.js', '/x/y/z.js', false)
```
Results in "../../a/b/c.js" in the sources output.

Is this an accident, or what is the rationale for it?

**Comment by johnjbarton** [comment]
Is it a problem?

**Comment by jmm** [comment]
Yes, the browser can't request the source file at the URL I expect. I expect the `sourceName` value to be passed through to output without being altered. I'm unaware of the logical basis for altering it based on the `outputName` (`file` in the source map) value.

**Comment by johnjbarton** [comment]
A changed the title, I guess that is what you mean.

**Comment by johnjbarton** [comment]
What will we do for the `sources` values for files imported by compiling the input called `sourceName`?

**Comment by jmm** [comment]
> What will we do for the `sources` values for files imported by compiling the input called `sourceName`?

I didn't even know traceur can do that. I suggest that before you make or consider any further changes to the API, you document the current API. There isn't even rudimentary documentation here: [Node.js API](https://github.com/google/traceur-compiler/wiki/Node.js-API).

Is that a CLI-only issue? Is there even any good reason to do that in Node instead of doing something like?:

```js
(new traceur.NodeCompiler({sourceMaps: 'inline'}))
  .compile(fs.readFileSync('/dir/file.js'), 'file.js', 'file.js')
```

If that's necessary, to support the CLI or because there's a good reason to have it in the API, then there'll probably need to be some optional param like `basedir` that is used the way `sourceRoot` was being used. As has already been discussed, it can not be `sourceRoot`. Some users may not care about exposing their local paths or care about the browser being able to request the source files from the server and won't bother providing the param.

**Comment by jmm** [comment]
If you need to figure out if there's a reason to have this at all, you should at least stop doing it for API calls (as opposed to CLI), or at the very least for API calls that include the `content` argument.
